src/Components/Button/Button.js

```diff
@@ -1,9 +1,15 @@
 import React from 'react';
+
+import { useMyCartContext } from "Context/GlobalContext"
+
 import { StyledButton } from './styled';
 
-const Button = ({ children }) => {
+const Button = ({ children, propsForFunction }) => {
+    
+    const { addToMyCart } = useMyCartContext();
+
     return(
-        <StyledButton>
+        <StyledButton onClick={()=>{addToMyCart(propsForFunction?.productId, propsForFunction?.data)}} >
```

Comment:
This is way to specific for a component that you'll be using multiple times. Instead of logically handling with `propsForFunction`, consider moving this outside of the component. It is better to receive the actual action you want this button to do. For example, you receive the `onClick` prop, and reuse that prop in here like `onClick={onClick}`. With this, the component doesn't need to know how to do the `click`, but rather just triggering it.